lib/git-shell-out-strategy.js

```diff
@@ -316,61 +317,31 @@ export default class GitShellOutStrategy {
   /**
    * File Status and Diffs
    */
-  async getStatusesForChangedFiles() {
-    const output = await this.exec(['status', '--untracked-files=all', '--ignore-submodules=dirty', '-z']);
-
-    const statusMap = {
-      'A': 'added',
-      'C': 'added', // technically copied, but we'll treat as added
-      'M': 'modified',
-      'D': 'deleted',
-      'U': 'modified',
-      '?': 'added',
-    };
-
-    const stagedFiles = {};
-    const unstagedFiles = {};
-    const mergeConflictFiles = {};
-    let statusToHead;
-
-    if (output) {
-      const lines = output.split('\0');
-      for (let i = 0; i < lines.length; i++) {
-        const line = lines[i];
-        if (line === '') { continue; }
-        const [_, x, y, rawFilePath] = line.match(/(.)(.)\s(.*)/); // eslint-disable-line no-unused-vars
-        const filePath = toNativePathSep(rawFilePath);
-        if (x === 'U' || y === 'U' || (x === 'A' && y === 'A')) {
-          // Skipping this check here because we only run a single `await`
-          // and we only run it in the main, synchronous body of the for loop.
-          // eslint-disable-next-line no-await-in-loop
-          if (!statusToHead) { statusToHead = await this.diffFileStatus({target: 'HEAD'}); }
-          mergeConflictFiles[filePath] = {
-            ours: statusMap[x],
-            theirs: statusMap[y],
-            file: statusToHead[filePath] || 'equivalent',
-          };
-          continue;
-        }
-        if (x === 'R') {
-          stagedFiles[filePath] = 'added';
-          stagedFiles[lines[++i]] = 'deleted';
-          continue;
-        }
-        if (y !== ' ') {
-          unstagedFiles[filePath] = statusMap[y];
-        }
-        if (x !== ' ' && x !== '?') {
-          stagedFiles[filePath] = statusMap[x];
+  async getStatusBundle() {
+    const output = await this.exec(['status', '--porcelain=v2', '--branch', '-z']);
+    const results = parseStatus(output);
+    const mapToNativePathSep = entries => {
```

Comment:
Not a *huge* deal, but can we extract this function so a new copy doesn't get created for every invocation of `getStatusBundle`?